tests/PHPStan/Analyser/data/dynamic-method-return-getsingle-conditional.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace DynamicMethodReturnGetSingleConditional;
+
+use function PHPStan\Testing\assertType;
+
+abstract class Foo
+{
+	/**
+	 * @return ($input is 1 ? true : false)
+	 */
+	abstract public function get(int $input): mixed;
+
+	public function doFoo(): void
+	{
+		assertType('bool', $this->get(0));
+		assertType('bool', $this->get(1));
+		assertType('bool', $this->get(2));
```

Comment:
do we expect a more precise type here?

```suggestion
		assertType('false', $this->get(0));
		assertType('true', $this->get(1));
		assertType('false', $this->get(2));
```

Reply:
No - there is return type extension that just returns `ParametersAcceptorSelector::selectSingle()` to verify that it provides the resulting union type instead of the conditional type.